File: Domain/Models/VisitLog.cs
using System;

namespace Domain.Models
{
    public class VisitLog
    {
        public Guid Id { get; set; }
        public string ParkName { get; set; }
        public string Title { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Notes { get; set; }
        public Guid ParkRef { get; set; }
        public AppUser AppUser { get; set; }
    }
}

File: client-app/src/app/models/visitLog.ts
export interface VisitLog {
    id: string;
    parkName: string;
    title: string;
    startDate: Date | null;
    endDate: Date | null;
    notes: string;
    parkRef: string;
}

export class VisitLog implements VisitLog {
    constructor(init?: VisitLogFormValues) {
      Object.assign(this, init);
    }
}

export class VisitLogFormValues {
    id?: string = undefined;
    parkName: string = '';
    title: string = '';
    startDate: Date | null = null;
    endDate: Date | null = null;
    notes: string = '';
    parkRef: string = '';

    constructor(visitLog?: VisitLogFormValues) {
        if (visitLog) {
          this.id = visitLog.id;
          this.parkName = visitLog.parkName;
          this.title = visitLog.title;
          this.startDate = visitLog.startDate;
          this.endDate = visitLog.endDate;
          this.notes = visitLog.notes;
          this.parkRef = visitLog.parkRef;
        }
      }
}